### ISSUE TITLE
libretro.play fails to build with gcc10

### ISSUE THREAD

**Issue by austinbutler** [NixOS/nixpkgs#108988]
**Describe the bug**

`libretro.play` is broken on master.

```
   62 | std::array<uint8, 0x20> Framework::HashUtils::ComputeHmacSha256(const void* key, size_t keySize, const void* data, size_t dataSize)
      |                                                                                                                                   ^
/build/play--884ae3b/deps/Framework/src/HashUtils.cpp: In function 'std::array<unsigned char, 32> Framework::HashUtils::ComputeHmacSha256(const void*, size_t, const void*, size_t)':
/build/play--884ae3b/deps/Framework/src/HashUtils.cpp:136:14: error: 'runtime_error' is not a member of 'std'
  136 |   throw std::runtime_error("Invalid result size");
      |              ^~~~~~~~~~~~~
make[2]: *** [Source/ui_libretro/Source/Framework/CMakeFiles/Framework.dir/build.make:225: Source/ui_libretro/Source/Framework/CMakeFiles/Framework.dir/__/__/src/HashUtils.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /build/play--884ae3b/deps/Framework/src/Config.cpp:4:
/build/play--884ae3b/deps/Framework/build_cmake/Framework/../../include/Config.h: In instantiation of 'static std::shared_ptr<_Tp> Framework::CConfig::CastPreference(const PreferencePtr&) [with Type = Framework::CConfig::CPreferenceInteger; Framework::CConfig::PreferencePtr = std::shared_ptr<Framework::CConfig::CPreference>]':
/build/play--884ae3b/deps/Framework/src/Config.cpp:82:34:   required from 'std::shared_ptr<_Tp> Framework::CConfig::FindPreference(const char*) [with Type = Framework::CConfig::CPreferenceInteger]'
/build/play--884ae3b/deps/Framework/src/Config.cpp:132:59:   required from here
/build/play--884ae3b/deps/Framework/build_cmake/Framework/../../include/Config.h:149:29: warning: comparison between 'enum Framework::CConfig::PREFERENCE_TYPE' and 'enum Framework::CConfig::CPreferenceInteger::<unnamed>' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wenum-compare-Wenum-compare8;;]
  149 |    if(preference->GetType() != Type::PREFERENCE_TYPE_ID)
      |       ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/play--884ae3b/deps/Framework/build_cmake/Framework/../../include/Config.h: In instantiation of 'static std::shared_ptr<_Tp> Framework::CConfig::CastPreference(const PreferencePtr&) [with Type = Framework::CConfig::CPreferenceBoolean; Framework::CConfig::PreferencePtr = std::shared_ptr<Framework::CConfig::CPreference>]':
/build/play--884ae3b/deps/Framework/src/Config.cpp:82:34:   required from 'std::shared_ptr<_Tp> Framework::CConfig::FindPreference(const char*) [with Type = Framework::CConfig::CPreferenceBoolean]'
/build/play--884ae3b/deps/Framework/src/Config.cpp:139:59:   required from here
/build/play--884ae3b/deps/Framework/build_cmake/Framework/../../include/Config.h:149:29: warning: comparison between 'enum Framework::CConfig::PREFERENCE_TYPE' and 'enum Framework::CConfig::CPreferenceBoolean::<unnamed>' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wenum-compare-Wenum-compare8;;]
/build/play--884ae3b/deps/Framework/build_cmake/Framework/../../include/Config.h: In instantiation of 'static std::shared_ptr<_Tp> Framework::CConfig::CastPreference(const PreferencePtr&) [with Type = Framework::CConfig::CPreferenceString; Framework::CConfig::PreferencePtr = std::shared_ptr<Framework::CConfig::CPreference>]':
/build/play--884ae3b/deps/Framework/src/Config.cpp:82:34:   required from 'std::shared_ptr<_Tp> Framework::CConfig::FindPreference(const char*) [with Type = Framework::CConfig::CPreferenceString]'
/build/play--884ae3b/deps/Framework/src/Config.cpp:146:58:   required from here
/build/play--884ae3b/deps/Framework/build_cmake/Framework/../../include/Config.h:149:29: warning: comparison between 'enum Framework::CConfig::PREFERENCE_TYPE' and 'enum Framework::CConfig::CPreferenceString::<unnamed>' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wenum-compare-Wenum-compare8;;]
/build/play--884ae3b/deps/Framework/build_cmake/Framework/../../include/Config.h: In instantiation of 'static std::shared_ptr<_Tp> Framework::CConfig::CastPreference(const PreferencePtr&) [with Type = Framework::CConfig::CPreferencePath; Framework::CConfig::PreferencePtr = std::shared_ptr<Framework::CConfig::CPreference>]':
/build/play--884ae3b/deps/Framework/src/Config.cpp:82:34:   required from 'std::shared_ptr<_Tp> Framework::CConfig::FindPreference(const char*) [with Type = Framework::CConfig::CPreferencePath]'
/build/play--884ae3b/deps/Framework/src/Config.cpp:153:56:   required from here
/build/play--884ae3b/deps/Framework/build_cmake/Framework/../../include/Config.h:149:29: warning: comparison between 'enum Framework::CConfig::PREFERENCE_TYPE' and 'enum Framework::CConfig::CPreferencePath::<unnamed>' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wenum-compare-Wenum-compare8;;]
[  7%] Linking C static library libbz2.a
make[1]: *** [CMakeFiles/Makefile2:426: Source/ui_libretro/Source/Framework/CMakeFiles/Framework.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  7%] Built target bz2
[  7%] Linking C static library libGLEW.a
[  7%] Built target glew_s
make: *** [Makefile:114: all] Error 2
builder for '/nix/store/v4fcj4mpmpfgfs2zjprgb9qky7z7r9pw-libretro-play-2020-03-06.drv' failed with exit code 2
error: build of '/nix/store/v4fcj4mpmpfgfs2zjprgb9qky7z7r9pw-libretro-play-2020-03-06.drv' failed
```

Suspecting it's due to `gcc10`, I did a simple find and replace for `stdenv` in `pkgs/misc/emulators/retroarch/cores.nix` and indeed it does build.

Nix uses https://github.com/libretro/Play- as the source, but it's a fork of https://github.com/jpd002/Play- that hasn't been updated since January 2020. A [fix for gcc10](https://github.com/jpd002/Play-/pull/918) was merged in June 2020.

I'm not sure how to use gcc9 _only_ for the `play` core, or if that's even the right solution. Alternatively we could maybe request that the libretro fork of play be updated.

**To Reproduce**

`nix-build -A libretro.play`

**Expected behavior**

The play core builds successfully.

**Notify maintainers**

@edwtjo @hrdinka @MP2E 

**Metadata**

```yml
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.5, NixOS, 21.03pre262347.257cbbcd3ab (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixos-21.03pre262347.257cbbcd3ab, home-manager"`
 - channels(austin): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - libretro.play
# a list of nixos modules affected by the problem
module:
```

**Comment by matthewbauer** [comment]
I don't think there's anything stopping us from just using the upstream Play. The libretro ones are usually better maintained, but in this case it looks like upstream is more recent.

**Comment by matthewbauer** [comment]
https://github.com/NixOS/nixpkgs/pull/110650 should fix

**Comment by thiagokokada** [comment]
Closed with #146295.

Not sure if it is working, but at least it is compiling now.
